cloud-functions/http_standardize_names/main.py

```diff
@@ -14,36 +17,20 @@ def http_standardize_names(request: flask.Request) -> flask.Response:
     :param data: Either a FHIR bundle or a FHIR-formatted JSON dict
     :return: The bundle or resource with names appropriately standardized
     """
+    content_type = "application/json"
     # Validate request header.
-    if request.headers.get("Content-Type") != "application/json":
-        logging.error("Header must inclue: 'Content-Type:application/json'.")
-        return {
-            "status": 400,
-            "summary": "Bad request",
-            "description": "Header must inclue: 'Content-Type:application/json'.",
-        }
-
-    try:
-        request_json = request.get_json(silent=False)
-    except AttributeError as error:
-        logging.error(error)
-
-        return {
-            "status": 400,
-            "summary": "Invalid request body",
-            "description": "Invalid JSON",
-        }
+    header_response = validate_request_header(request, content_type)
 
     # Check that the request body contains a FHIR bundle or resource.
-    if request_json.get("resourceType") is None:
-        error_message = (
-            "FHIR Resource Type not specified. "
-            + "The request body must contain a valid FHIR bundle or resource."
-        )
-        logging.error(error_message)
-        return {"status": 400, "summary": "Bad request", "description": error_message}
-
-    # Perform the name standardization
-    response = standardize_names(request_json)
-
-    return response
+    if header_response.status_code != 400:
+        body_response = validate_fhir_bundle_or_resource(request)
+        if body_response.status_code != 400:
+            request_json = request.get_json(silent=False)
+            # Perform the name standardization
+            final_response = standardize_names(request_json)
+        else:
+            return body_response
+    else:
+        return header_response
+
+    return final_response
```

Comment:
Are we being inconsistent here with our return type? Looks like we always return a `flask.Response` object except when everything works properly. I believe `final_response` is a `dict`.

Reply:
I was thinking about this.  Should we construct a flask.Response that contains the json result that is passed back in?  That's where my head is at for this.

Reply:
That's what I was thinking.